lighty-modules/lighty-aaa-aggregator/lighty-aaa/src/test/java/io/lighty/aaa/ShiroInitializationTest.java

```diff
@@ -0,0 +1,115 @@
+    @Mock
+    private DataBroker bindingDataBroker;
+    @Mock
+    private ReadTransaction readTransaction;
+    @Mock
+    private CredentialAuth<PasswordCredentials> credentialAuth;
+    private AAALighty aaaLighty;
+
+    @BeforeClass
+    public void init() {
+        // Initialize the mock objects
+        MockitoAnnotations.initMocks(this);
+
+        // Set up mocks for some datastore reads
+        when(bindingDataBroker.newReadOnlyTransaction()).thenReturn(readTransaction);
+        final AaaEncryptServiceConfig aaaEncryptServiceConfig = new AaaEncryptServiceConfigBuilder().build();
+        when(readTransaction.read(LogicalDatastoreType.CONFIGURATION, MdsalUtils.getEncryptionSrvConfigIid()))
+                .thenReturn(FluentFutures.immediateFluentFuture(Optional.of(aaaEncryptServiceConfig)));
+
+        final KeyStores keyStores = new KeyStoresBuilder().build();
+        when(readTransaction.read(LogicalDatastoreType.CONFIGURATION, InstanceIdentifier.create(KeyStores.class)))
+                .thenReturn(FluentFutures.immediateFluentFuture(Optional.of(keyStores)));
+
+        final SslData sslData = new SslDataBuilder()
+                .setBundleName(BUNDLE_NAME).build();
+        when(readTransaction.read(LogicalDatastoreType.CONFIGURATION, KeyStoresDataUtils.getSslDataIid(BUNDLE_NAME)))
+                .thenReturn(FluentFutures.immediateFluentFuture(Optional.of(sslData)));
+    }
+
+    // Test that the AAALighty object can be stopped after failing to initialize
+    @Test
+    public void testStopProcedureWithFailedInitialization() {
+        // Create an AAALighty object with mocked dependencies
+        this.aaaLighty = new AAALighty(bindingDataBroker, credentialAuth, server, AAA_CONFIGURATION);
+        // Ensure that the object was created successfully
+        assertNotNull(aaaLighty);
+        // Expect an Exception to be thrown when trying to initialize the object
+        expectThrows(Exception.class, () -> aaaLighty.initProcedure());
+    }
+
+    // Test that the AAALighty object can be successfully initialized and stopped
+    @Test
+    public void testSuccessfulInitialization() throws InterruptedException {
+        // set CertificateManager
+        AAA_CONFIGURATION.setCertificateManager(CertificateManagerConfig.getDefault(bindingDataBroker));
+
+        // Create a LightyServerBuilder object
+        final LightyServerBuilder serverBuilder = new LightyServerBuilder(
+                new InetSocketAddress("localhost/127.0.0.1", 8182));
+
+        // Create an AAALighty object
+        this.aaaLighty = new AAALighty(bindingDataBroker, null, serverBuilder, AAA_CONFIGURATION);
+        // Ensure that the object was created successfully
+        assertNotNull(aaaLighty);
+
+        // Initialize the object and ensure that initialization was successful
+        final boolean initProcedure = aaaLighty.initProcedure();
+        assertTrue(initProcedure);
+
+    }
+
+    @AfterMethod
```

Comment:
A good convention to follow is to have the "AfterMethod", "After", "BeforeMethod", etc. be the first methods in a test. Could you please move it below the "init()" method?

Reply:
done